src/EventStore.Core/ClusterVNode.cs

```diff
@@ -906,6 +932,74 @@ public class ClusterVNode :
 
+		public void Handle(ClientMessage.ReloadConfig message) {
+			if (Interlocked.CompareExchange(ref _reloadingConfig, 1, 0) != 0) {
+				Log.Information("The node's configuration reload is already in progress");
+				return;
+			}
+
+			Task.Run(() => {
+				try {
+					Log.Information("Reloading the node's configuration");
+					var options = _vNodeSettings.LoadConfigFunc();
+					ReloadCertificates(options);
+				} catch (Exception exc) {
+					Log.Error(exc, "An error has occurred while reloading the configuration");
+				} finally {
+					Interlocked.Exchange(ref _reloadingConfig, 0);
+				}
+			});
+		}
+
+		private void ReloadCertificates(ClusterNodeOptions options) {
+			var currentCertificate = _certificateSelector();
+			var currentTrustedRootCerts = _trustedRootCertsSelector();
+
+			if (currentCertificate == null || currentTrustedRootCerts == null) {
+				Log.Information("Skipping reload of certificates since TLS is disabled.");
+				return;
+			}
+
+			X509Certificate2 certificate;
+			if (!string.IsNullOrWhiteSpace(options.CertificateStoreLocation)) {
+				var location = CertificateLoader.GetCertificateStoreLocation(options.CertificateStoreLocation);
+				var name = CertificateLoader.GetCertificateStoreName(options.CertificateStoreName);
+				certificate = CertificateLoader.FromStore(location, name, options.CertificateSubjectName,
+					options.CertificateThumbprint);
+			} else if (!string.IsNullOrWhiteSpace(options.CertificateStoreName)) {
+				var name = CertificateLoader.GetCertificateStoreName(options.CertificateStoreName);
+				certificate = CertificateLoader.FromStore(name, options.CertificateSubjectName,
+					options.CertificateThumbprint);
+			} else if (options.CertificateFile.IsNotEmptyString()) {
+				certificate = CertificateLoader.FromFile(
+					options.CertificateFile,
+					options.CertificatePrivateKeyFile,
+					options.CertificatePassword);
+			} else {
+				throw new InvalidConfigurationException(
+					"A certificate is required but none was provided in the configuration.");
+			}
+
+			var trustedRootCerts = new X509Certificate2Collection();
+			if (!string.IsNullOrEmpty(options.TrustedRootCertificatesPath)) {
+				foreach (var (fileName, cert) in CertificateLoader.LoadAllCertificates(options.TrustedRootCertificatesPath)) {
+					trustedRootCerts.Add(cert);
+					Log.Information("Trusted root certificate file loaded: {file}", fileName);
+				}
+				if(trustedRootCerts.Count == 0)
+					throw new InvalidConfigurationException($"No trusted root certificate files were loaded from the specified path: {options.TrustedRootCertificatesPath}");
+			} else {
+				throw new InvalidConfigurationException(
+					$"{nameof(options.TrustedRootCertificatesPath)} was not specified in the configuration.");
+			}
+
+			//no need for a lock here since reference assignment is atomic
+			_certificate = certificate;
+			_trustedRootCerts = trustedRootCerts;
+
+			Log.Information("Certificates have been successfully reloaded");
```

Comment:
It might be nice to include the thumbprint of both the old and the new certificates here?

Reply:
I've added a new commit: f63f989b51c6172af25649bd53d9e3fe6ca291e9 thanks